azure/Guidelines.md

```diff
@@ -84,84 +82,103 @@ https://<tenant-id>-<service-defined-root>.<service>.azure.net
 
-#### Property name has changed
+| Property change        | Input        | Output       |
+|:-----------------------|:------------:|:------------:|
+| Remove a property      | Breaking     | Breaking     |
+| Add optional property  | Evolutionary | Breaking     |
+| Add required property  | Breaking     | Breaking     |
+| Data type change       | Breaking     | Breaking     |
+| Format change          | Breaking     | Breaking     |
+| Integer widens         | Evolutionary | Breaking     |
+| Integer narrows        | Breaking     | Evolutionary |
+| Add new value to enum  | Evolutionary | Breaking     |
+| Remove value from enum | Breaking     | Breaking     |
+| Optional to required   | Breaking     | Breaking     |
+| Required to optional   | Evolutionary | Breaking     |
 
-Note that this is implied by the requirement that adding/and removing properties are breaking changes, but in some ways worse, since it leads to the possibility of reusing a property name. Even with an API version change, this change is discouraged because it creates documentation and cognitive challenges.
+Breaking changes require prior approval of the Azure REST API review board and approval through the [Azure Global Breaking Change Policy][7]. In the case of deprecation, follow the [Azure Global Retirement Policy][7].  If the service is using SemVer for versioning, breaking changes constitute a major version change.
 
-#### Property type has changed
+Evolutionary changes do not require prior approval (but still need a version bump).  If the service is using SemVer for versioning, evolutionary changes constitute a minor version change.
 
-Property `foo was a boolean in v1 but is changed to a string. A client using the existing API version tries to set it as a boolean, but the service will fail since its now expecting a string. So, the API version must be updated.
+#### Changing the API without changing the version
 
-#### Property default value has changed
+There are a limited set of situations where changing the API is permissable without a version bump:
 
-If a property is optional and the service provides a default value, changing that default requires an updated API version.
+1. Adding a new (optional) value to an extensible enum.
+2. Adding a new (optional) query parameter to adjust the output (for example, adding filter options to a list operation).
+3. Adding optional computed read-only output values that are generated based on the new (optional) query parameters.
```

Comment:
IN addition, as mentioned on the call, having #2 and #3 will result in sub-optimal api design. These are service generated properties, and now customer will have to understand which query param to use to get a particular value. And it will also result in an explosion of query params if for every new read only property, a new query param gets introduced.

Reply:
@adrianhall: In addition to the wording you're going to add about not allowing this for multi-cloud services, I also think wording should be added that the service must be updated in ALL datacenters before the new query parameter is advertised to customers. This removes the race condition of a customer passing the new parameter against a version of the service that doesn't support the query parameter.

Reply:
Ack - added language around this.